rfcs/003-randomness.md

```diff
@@ -0,0 +1,151 @@
+# RFC: Randomness
+
+## Abstract
+
+This document covers existing randomness methods in Polkadot and discusses which
+would be appropriate to the Polka Storage project.
+
+## Introduction
+
+The Polka Storage project requires randomness as a form of creating challenges
+for Storage Providers, with the aim of stopping them from discarding the files
+they are supposed to store, by pre-computing proofs for upcoming challenges.
+Achieving this randomness is not trivial — for example, we cannot allow the
+Storage Provider to generate the randomness alone since they may be dishonest
+and generate values that allow them to fake the storage proofs; furthermore by
+their very nature, blockchains must be deterministic and as such, randomness
+methods achieved on-chain are not suitable for all use cases.
+
+## Terminology
+
+* SP — Storage Provider
+* VRF — Verifiable Randomness Function
+* PoRep — Proof of Replication
+* PoSt — Proof of Spacetime
+* drand — Distributed Randomness Beacon
+
+## The Problem
+
+The Polka Storage Parachain SPs generate proofs that they are, in fact, storing
+the files they committed to store. These proofs are PoRep and PoSt and they sit
+at the core of the system — breaking them means breaking the system, allowing
+storage to be proven without any actual storage being done.
+
+Randomness plays a different role in each proof; for PoRep, the randomness plays
+a crucial role in the generation of a unique sector, avoiding the possibility
+that the storage is outsourced or duplicated to another provider — meaning
+two providers share the same storage space, keeping a single copy of the data;
+for PoSt, SPs receive random challenges, these ensure that the SP cannot
+pre-compute proof values, allowing them to discard the data while faking storage.
```

Comment:
This sentence is kinda hard to read. I don't really understand what you mean with `meaning two providers share the same storage space, keeping a single copy of the data`, or rather I do not see the value of this sentence.

Reply:
```mermaid
flowchart LR
    provider1 --> storage
    provider2 --> storage
```